mobile-app/src/app/(auth)/emailsignup.tsx

```diff
@@ -28,8 +32,24 @@ export default function emailsignup() {
     console.log("Sign Up Attempted");
     createUserWithEmailAndPassword(auth, data.email, data.password).then(
       (success) => {
-        handleDatabaseSignUp();
-        resetNavigationStack("/");
+        const dataCopy = { ...data };
+        handleDatabaseSignUp(dataCopy).then(() => {
+          if (auth.currentUser) {
```

Comment:
instead of `auth.currentUser`, use the `userCredential` passed in side the `then()` (the thing named success). 
Then i don' think u would need to do the typescript if check
ref -> 
https://firebase.google.com/docs/auth/web/password-auth#create_a_password-based_account